### PR TITLE
fix(ui): fetch-stream SSE for authenticated Portal hosts

### PR DESCRIPTION
**Pull request by cidrblock**
## Summary
- Replace `EventSource` in `@apme/ui-workflow` with **fetch + ReadableStream** SSE via `adapter.fetch`, so hosts that authenticate with Backstage Bearer (Portal catalog proxy) receive live operation updates.
- Extract `parseSseChunk` / `readSseStream` and shared `applyOperationSseEvent` handlers; add vitest coverage.
- Force-enable `refresh()` after `startScan` so a stale `attachOp=false` closure cannot no-op the post-start poll (related Portal stuck spinner).

**Why:** Portal `/operation/events` was 401ing under `EventSource` (no `Authorization` header). The UI kept the first “Cloning repository…” snapshot while the Gateway had already reached `assessed`.

Native SPA keeps working: its adapter `fetch` is still plain `fetch` through the Vite `/api/v1` proxy.

## Test plan
- [x] `npm test` in `frontend/` (115 tests, including new `sseFetch` cases)
- [x] `npm run typecheck` in `frontend/`
- [x] `tox -e lint`
- [x] `tox -e unit`
- [ ] Local Portal smoke: Quality → Scan → progress advances past Cloning without poll workaround
- [ ] Native SPA: project Session still streams progress via Vite proxy

## Summary by CodeRabbit

* **New Features**
  * Added authenticated Server-Sent Events streaming for real-time project operation updates.
  * Live events now update project state including status, progress, proposals/findings, AI triage candidates, results/approvals, pull request links, and errors.
  * Introduced SSE parsing and stream-reading utilities, now available via the package’s public API.
* **Bug Fixes**
  * Improved the operation-state update lifecycle with safer abort handling and more reliable reconnection behavior.
* **Tests**
  * Added automated coverage for SSE chunk parsing, ordered streaming delivery, decoder flush behavior, abort edge cases, and state updates from events.

### CODE REVIEW

File: frontend/packages/ui-workflow/src/hooks/useProjectOperationState.ts [hunk 1]
```diff
@@ -508,63 +438,133 @@ export function useProjectOperationState(
...
+      if ((!enabled && !force) || !projectId) {
+        if (!force) setState(null);
         return;
       }
-      setState(s);
-      if (!TERMINAL_STATUSES.has(s.status)) {
-        connect();
+      try {
+        const s = await fetchProjectOperationState(projectId);
+        if (!mountedRef.current && !force) return;
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply frontend/packages/ui-workflow/src/hooks/useProjectOperationState.ts]: Accepted. `force` now only bypasses `enabled`; unmounted components always return before `setState`. Fixed in e99e078.

File: frontend/packages/ui-workflow/src/hooks/useProjectOperationState.ts [hunk 2]
```diff
@@ -508,63 +438,133 @@ export function useProjectOperationState(
...
+        if (ac.signal.aborted) return;
+        if (!mountedRef.current || gen !== connectGenRef.current) return;
+        setConnected(false);
+        reconnectTimer.current = setTimeout(() => {
+          if (mountedRef.current && gen === connectGenRef.current) {
+            connect();
+          }
+        }, 3000);
+        void err;
+      }
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply frontend/packages/ui-workflow/src/hooks/useProjectOperationState.ts]: Accepted. Switched to bare `catch { ... }` and dropped `void err`. Fixed in e99e078.

[the remark this answers is not in the document]
Comment (cidrblock) [reply frontend/packages/ui-workflow/src/hooks/useProjectOperationState.ts]: Accepted. Reconnect is gated on `statusRef` ∈ `TERMINAL_STATUSES` (Gateway closes after terminal snapshot; `assessed` stays live). Error path now uses exponential backoff capped at 30s. Fixed in e99e078.

File: frontend/packages/ui-workflow/src/api/sseFetch.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        onEvent(ev);
+      }
+    }
+    // Flush any final frame that ended without a trailing blank line.
+    if (buffer.trim()) {
+      const { events } = parseSseChunk(`${buffer}\n\n`);
+      for (const ev of events) {
+        onEvent(ev);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply frontend/packages/ui-workflow/src/api/sseFetch.ts:98]: Accepted. Final buffer flush is skipped when `signal.aborted` so teardown cannot emit a post-cancel event. Added a regression test. Fixed in d3b7207.

File: frontend/packages/ui-workflow/src/api/sseFetch.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    // Flush any final frame that ended without a trailing blank line.
+    // Skip on abort — cleanup/unmount must not emit post-cancel events.
+    if (!signal?.aborted && buffer.trim()) {
+      const { events } = parseSseChunk(`${buffer}\n\n`);
+      for (const ev of events) {
+        onEvent(ev);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (cidrblock) [reply frontend/packages/ui-workflow/src/api/sseFetch.ts:98]: Accepted. After the read loop we now call `decoder.decode()` (no `stream: true`) before the final frame flush, still skipped on abort. Regression test covers a dangling multi-byte lead byte at EOS. Fixed in 8b3d837.
